### ISSUE TITLE
Detect close loops in dependencies

### ISSUE THREAD

**Issue by Phibedy** [conan-io/conan#522]
It might be useful to detect closed loops in the  dependency tree. Of course it is not necessary as atm python just reaches it recursion limit
```
ERROR: gtest/1.8.0@lms/stable: Error in config, config_options or configure() method, line 125
	new_reqs = down_reqs.copy()
	maximum recursion depth exceeded in cmp
```

**Comment by memsharded** [comment]
Agree, dependency cycles are not allowed by design (as in other package managers), but some detection that reports it, could help. Thanks for the suggestion, will try to add it.

**Comment by monsdar** [comment]
This could be related, the thought just popped up by reading this:
Our CI was taking forever when we accidentally had cyclic dependencies in our conanfiles. It blocked the runner for a good hour before I saw that and cancelled the build job.

I'm not sure why Python was not running into the above mentioned recursion limit.

However to avoid things like that it of course would be better to have some kind a error message and exit the install with an errorcode.

**Comment by memsharded** [comment]
Yes, not sure why python wouldn't hit the limit and crashed.
I have just submitted a PR to detect and report loops, it should help with this issue.

**Comment by memsharded** [comment]
Implemented in #529, will be released in 0.14
